lib/modules/hideSidebar.js

```diff
@@ -0,0 +1,116 @@
+/* @flow */
+import { $ } from '../vendor';
+import { Module } from '../core/module';
+import { i18n } from '../environment';
+import * as Menu from './menu';
+import {
+	CreateElement
+} from '../utils';
+
+export const module: Module<*> = new Module('hideSidebar');
+
+module.moduleName = 'hideSidebarName';
+module.category = 'productivityCategory';
+module.description = 'hideSidebarTabDesc';
+
+let hideSidebar;
+let propToChange;
+let valueToChangeTo;
+let valueToChangeFrom;
+let $side;
+let $content;
+let contentOffset;
+let headerHeight;
+let documentHalf;
+let sideContent;
+let sideWidth;
+let rulesCheck = ['reddiquette', 'rules'];
+
+const toggleSidebar = () => {
+
+	if (!$content.data('sidebar-hidden')) {
+		let check = true;
+		if (rulesCheck.some(function (v) { return sideContent.indexOf(v) >= 0; })) {
+			check = confirm("Have you read the rules?");
+		}
+		if (check) {
+			doHideSidebar();
+		}
+	} else {
+		doShowSidebar();
+	}
+}
+
+function recursiveEach($element) {
+	$element.children().each(function () {
+		let $this = $(this);
+		let thisOffset = $this.offset();
+		if ($this.css("position") == "absolute" &&
+			thisOffset.top >= headerHeight &&
+			thisOffset.top <= contentOffset.top &&
+			thisOffset.left < documentHalf
+		) {
+			$this.css("visibility", "visible");
+		}
+		recursiveEach($this);
+	});
+
+}
+
+const doHideSidebar = () => {
+	$side.css({
+		"visibility": "hidden",
+		"width": 0
+	});
+	recursiveEach($side);
+	$content.css(`${propToChange}-right`, valueToChangeTo);
+	hideSidebar.innerText = "show sidebar";
+	$content.data('sidebar-hidden', true);
+}
+
+const doShowSidebar = () => {
+	$side.css({
+		"visibility": "visible",
+		"width": sideWidth + "px"
+	});
+	$content.css(`${propToChange}-right`, valueToChangeFrom);
+	hideSidebar.innerText = "hide sidebar";
+	$content.data('sidebar-hidden', false);
+}
+
+module.go = () => {
+	$side = $(".side");
+	$content = $(".content[role=main]");
+	documentHalf = $(document).width() / 2;
+	contentOffset = $content.offset();
+	headerHeight = $("#header[role=banner]").height();
+	sideContent = $side.text().toLowerCase();
+	sideWidth = $side.width();
+
+	let padding = $content.css("padding").split(" ");
```

Comment:
`const` -- you're never going to change the value of `padding`. same with `margin`

Reply:
It appears you expect `.css('padding')` to return all four values at once, which is an implementation detail of jQuery that I wouldn't trust. Instead, you should specifically call `.css('padding-top')` and `.css('padding-left')` etc to get the exact values you need.  Additionally, you should put them in more descriptively named variables like `paddingTop`, `paddingLeft`.

Reply:
Re: ```const``` - sure.
Re: ```.css('padding')``` I did some investigation, and while running the function several times has a [performance hit](https://jsperf.com/res-jquery-shorthand-test/1) I looked up how [jQuery does the calculation](https://github.com/jquery/jquery/blob/master/src/css/var/getStyles.js#L15) and did a test of my own:

```
<!DOCTYPE html>
<html lang="en">
<head>
	<meta charset="UTF-8">
	<meta name="viewport" content="width=device-width, initial-scale=1.0">
	<meta http-equiv="X-UA-Compatible" content="ie=edge">
	<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.2.1/jquery.min.js"></script>
	<title></title>

	<style>
		#test {
			padding-left: 10px;
			padding-top: 15px;
			/*margin-left: 20px;*/
		}
	</style>

</head>
<body>
	
	<div id="test"></div>

	<script>
		
	$(function() {

		var $t = $("#test");
		var pr = $t.css("padding-right");
		var pl = $t.css("padding-left");
		var mr = $t.css("margin-right");
		var ml = $t.css("margin-left");

		console.log(pr, pl, mr, ml);

	});

	$(function() {

		var $t = $("#test");
		var ps = $t.css("padding").split(" ");
		var ms = $t.css("margin").split(" ");

		// If no margin set, returns undefined, which will break parseInt
		console.log(ps[1], ps[3], ms[1], ms[3]);

	});
	</script>

</body>
</html>
```

Coupled with this from the [jQuery docs](http://api.jquery.com/css/):

> Retrieval of shorthand CSS properties (e.g., margin, background, border), although functional with some browsers, is not guaranteed.

It shows you're right - I'll make it more explicit.